### PR TITLE
Add section about optional force unwrapping

### PR DESCRIPTION
**Pull request by 3lvis**

### CODE REVIEW

File: BEST_PRACTICES.md [hunk 1]
```diff
@@ -253,6 +254,19 @@ Views should be layout using Apple's Auto Layout. No third-party frameworks are
 
 Old style layout is still an option for when Auto Layout is not available.
 
+## Optional Force Unwrapping
+
+When something that shouldn't return an optional, returns an optional you have two options:
+
+**A)** Pass the error to the caller (throw) .
```
Comment (marijnschilling) [inline BEST_PRACTICES.md]: You could add _when_ to do this: 

`Pass the error to the caller (throw) , _if the app can recover._`
`Crash, _when the app can't recover_`

File: BEST_PRACTICES.md [hunk 2]
```diff
@@ -253,6 +254,18 @@ Views should be layout using Apple's Auto Layout. No third-party frameworks are
 
 Old style layout is still an option for when Auto Layout is not available.
 
+## Optional Force Unwrapping
+
+When something that shouldn't return an optional, returns an optional you have two options:
+
+**A)** Pass the error to the caller (throw): This is the ideal way of handling it, specially if there's the possibility of the app to recover. It's pretty straight forward, you create an `Error` or `NSError` with the right status code, some information about the error in the info field and you move on.
+
+For example, lets say you have a method that returns a filename for a photo, in theory all the photos should have a filename, so, the return value shouldn't be an optional. In this case you could pick returning a `String` and using a `fatalError` for when the filename is not found, but is ideal that you use `throws` instead and let the method called to decide if they either go for crashing the app, or going for just excluding the photo with the missing filename.
```
Comment (marijnschilling) [inline BEST_PRACTICES.md]: This example might be clearer when you only give the example of what to do in this case. So leave out `In this case you could pick returning a `String` and using a `fatalError` for when the filename is not found, but is ideal that you`

So just say: `throw an error is this case and let the method called to decide...`

**Comment by 3lvis** [comment]
@marijnschilling updated!

**Comment by marijnschilling** [comment]
🆒🚀

File: BEST_PRACTICES.md [hunk 3]
```diff
@@ -253,6 +254,18 @@ Views should be layout using Apple's Auto Layout. No third-party frameworks are
 
 Old style layout is still an option for when Auto Layout is not available.
 
+## Optional Force Unwrapping
+
+When something that shouldn't return an optional, returns an optional you have two options:
+
+**A)** Pass the error to the caller (throw): This is the ideal way of handling it, specially if there's the possibility of the app to recover. It's pretty straight forward, you create an `Error` or `NSError` with some information about the error, and throw it. [More information about throwing here.](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/ErrorHandling.html)
```
Comment (elland) [inline BEST_PRACTICES.md:261]: > specially if there's the possibility of the app to recover

specially if it's possible for the app to recover. 😄

File: BEST_PRACTICES.md [hunk 4]
```diff
@@ -253,6 +254,18 @@ Views should be layout using Apple's Auto Layout. No third-party frameworks are
...
+
+For example, lets say you have a method that returns a filename for a photo, in theory all the photos should have a filename, so, the return value shouldn't be an optional. Throw an error in this case and let the method caller decide if they either go for crashing the app, or for just excluding the photo with the missing filename.
```
Comment (elland) [inline BEST_PRACTICES.md:263]: "for example" and "lets say…" both have the same function. We use one or the other. 👍

File: BEST_PRACTICES.md [hunk 5]
```diff
@@ -253,6 +254,18 @@ Views should be layout using Apple's Auto Layout. No third-party frameworks are
...
+
+**B)** Crash: is a bit more tricky, here you could use `fatalError()` to provide more information about the crash, or just force unwrap it.
```
Comment (elland) [inline BEST_PRACTICES.md:265]: "it's a bit more tricky". Missing subject :)

File: BEST_PRACTICES.md [hunk 6]
```diff
@@ -253,6 +254,18 @@ Views should be layout using Apple's Auto Layout. No third-party frameworks are
...
+
+In cases where the crash is obvious such as when dequeuing cells and casting them to your specific cell class, force unwrapping is preferred, because if you can't dequeue the cell, or can't a cell for the specific class, there's no other way to recover than to show an empty screen, which is not preferable.
```
Comment (elland) [inline BEST_PRACTICES.md:267]: > obvious such

missing comma, `obvious, such`

Comment (elland) [reply BEST_PRACTICES.md:267]: > or can't a cell for the specific class

accidentally a word? 😋

**Comment by elland** [comment]
Some grammar nazism 👍

**Comment by 3lvis** [comment]
@Elland this kind of feedback for an already merged PR is not very useful, it will most likely be ignored unless you make a PR fixing the typos.

**Comment by elland** [comment]
@3lvis there was a request for me to review it, maybe it shouldn't be merged before it happened? 🤔

**Comment by 3lvis** [comment]
@Elland So far we haven't required that all the team give thumbs up before merging a PR, and I don't think we should start doing that, if you feel strongly about it I encourage you to make a PR to discuss it.

Personally, if I assign more than one person to the PR then it means anyone who's available can review it.

**Pull request merged**